src/transformers/models/idefics/__init__.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import TYPE_CHECKING
+
+from ...utils import OptionalDependencyNotAvailable, _LazyModule, is_torch_available, is_vision_available
+
+
+_import_structure = {"configuration_idefics": ["IDEFICS_PRETRAINED_CONFIG_ARCHIVE_MAP", "IdeficsConfig"]}
+
+try:
+    if not is_vision_available():
+        raise OptionalDependencyNotAvailable()
+except OptionalDependencyNotAvailable:
+    pass
+else:
+    _import_structure["image_processing_idefics"] = ["IdeficsImageProcessor"]
+    _import_structure["processing_idefics"] = ["IdeficsProcessor"]
```

Comment:
The Processor should be outside of the vision part, as its modeling file does not require `is_vision_available` (and for consistency with other similar processors).

Reply:
sure, but then you have to also fix:

```
src/transformers/models/flava/__init__.py:    _import_structure["processing_flava"] = ["FlavaProcessor"]
src/transformers/models/vilt/__init__.py:    _import_structure["processing_vilt"] = ["ViltProcessor"]
```

I'm pretty sure I didn't add it, and that it was the cloning script that added it via vilt as a source

Reply:
Thinking about it more - while technically the processor indeed doesn't require vision directly, the Idefics processor requires image processing, as it automatically loads `image_processor_class` - so it requires vision - but at instantiation time, rather than import time.

But if you believe that the key is what the file imports is the important part and since it imports nothing that has vision then it shouldn't require vision.

But whatever you think is preferable works for me.